src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -230,6 +234,30 @@ private void createXmlTest(ITestContext context, List<ITestNGMethod> methods, Xm
   }
 
   /**
+   * 
+   * @param srcXmlTest
+   * @param method the method we want to find its parameters
+   * @return local parameters belong to one test method.
+   */
+  private Map<String, String> findMethodLocalParameters(XmlTest srcXmlTest, ITestNGMethod method) {
+      Object[] instances= method.getInstances();
+      Class clazz= instances == null || instances.length == 0 || instances[0] == null
+          ? method.getRealClass()
+          : instances[0].getClass();
```

Comment:
I think a comment is missing here: why not using `method.getRealClass()` everytime?